stellarphot/core.py

```diff
@@ -62,3 +64,307 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+
+    star_id: `astropy.table.Column`
+        A column of unique object ids.
+
+    xcenter: `astropy.table.Column`
+        A column of the central x position of this object.
+
+    ycenter: `astropy.table.Column`
+        A column of the central y position of this object.
+    """
+
+    # Define columns in the photo_table and provide information about their type, units,
+    # and what attribute name to map that column into (each row of array is
+    # name, dtype, unit, attr_name.
+    phot_descript = np.array([['id', '<i8', None, 'id'],
+                            ['xcenter', '<f8', u.pix, 'xcenter'],
+                            ['ycenter', '<f8', u.pix, 'ycenter'],
+                            ['aperture_sum', '<f8', u.adu, 'ap_sum'],
+                            ['annulus_sum', '<f8', None, 'ann_sum'],
+                            ['RA', '<f8', u.deg, 'ra'],
+                            ['Dec', '<f8', u.deg, 'dec'],
+                            ['sky_per_pix_avg', '<f8', u.adu, 'spp_avg'],
+                            ['sky_per_pix_med', '<f8', u.adu, 'spp_med'],
+                            ['sky_per_pix_std', '<f8', u.adu, 'spp_std'],
+                            ['fwhm_x', '<f8', None, None],
+                            ['fwhm_y', '<f8', None, None],
+                            ['width', '<f8', None, 'fwhm'],
+                            ['aperture', '<f8', u.pix, 'ap_rad'],
+                            ['aperture_area', '<f8', None, 'ap_area'],
+                            ['annulus_inner', '<f8', u.pix, 'ann_in'],
+                            ['annulus_outer', '<f8', u.pix, 'ann_out'],
+                            ['annulus_area', '<f8', None, 'ann_area'],
+                            ['exposure', '<f8', u.s, 'exposure'],
+                            ['date-obs', '<U23', None, 'date'],
+                            ['night', '<i8', None, 'night'],
+                            ['aperture_net_flux', '<f8', u.adu, 'ap_netflux'],
+                            ['BJD', '<f8', None, 'bjd'],
+                            ['mag_inst', '<f8', None, 'mag_inst'],
+                            ['airmass', '<f8', None, 'airmass'],
+                            ['filter', '<U2', None, 'filter'],
+                            ['file', '<U38', None, None],
+                            ['star_id', '<i8', None, 'star_id'],
+                            ['mag_error', '<f8', u.adu**-1, 'mag_err'],
+                            ['noise', '<f8', None, 'noise'],
+                            ['noise-aij', '<f8', None, 'noise_aij'],
+                            ['snr', '<f8', u.adu, 'snr']])
+
+    def __init__(self, observatory, camera, filter_map, data = None):
+        # Set metavariables
+        self.observatory = observatory
+        self.camera = camera
+        self.filter_map = filter_map
+
+        # Handle data table
+        super().__init__(self.phot_descript, data)
+
+
+    def update_filters(self):
```

Comment:
It seems like this should be called in the `__init__`

Reply:
Also, I think we'll need to allow for the possibility that there is not a corresponding filter, eg `540_bp40`.

Reply:
Well, the way it currently works, those filters will simply not be changed.

Reply:
> It seems like this should be called in the `__init__`

Wan't sure if you wanted to force the issue, also, it should be possible to create a BLANK table using these classes, and then fill the table, in which case, you would want to call update_filters() later.

Reply:
> Also, I think we'll need to allow for the possibility that there is not a corresponding filter, eg `540_bp40`.

Well, the way the code currently currently works, if the filter is not mentioned in the filter_map, the filter will simply not be changed.  So the filter_map should only be for those filters that *do* map to standard AAVSO names.

Reply:
> Well, the way the code currently currently works, if the filter is not mentioned in the filter_map, the filter will simply not be changed. So the filter_map should only be for those filters that do map to standard AAVSO names.

Good point...

Reply:
> Wan't sure if you wanted to force the issue, also, it should be possible to create a BLANK table using these classes, and then fill the table, in which case, you would want to call update_filters() later.

I think we do want to force the issue but you are right we also need to update if data is added later. Not 100% sure we can enforce that, though